Mathlib/LinearAlgebra/RootSystem/Reduced.lean

```diff
@@ -153,16 +168,65 @@ lemma pairing_neg_one_neg_four_iff' (h2 : IsSMulRegular R (2 : R)) :
     P.pairing i j = -1 ∧ P.pairing j i = -4 ↔ P.root j = (-2 : R) • P.root i := by
   simpa [neg_smul, ← neg_eq_iff_eq_neg] using P.pairing_one_four_iff' i (P.reflection_perm j j) h2
 
+/-- See also `RootPairing.pairingIn_one_four_iff`. -/
 @[simp]
 lemma pairing_one_four_iff [NoZeroDivisors R] :
     P.pairing i j = 1 ∧ P.pairing j i = 4 ↔ P.root j = (2 : R) • P.root i :=
   P.pairing_one_four_iff' i j <| smul_right_injective R two_ne_zero
 
+/-- See also `RootPairing.pairingIn_neg_one_neg_four_iff`. -/
 @[simp]
 lemma pairing_neg_one_neg_four_iff [NoZeroDivisors R] :
     P.pairing i j = -1 ∧ P.pairing j i = -4 ↔ P.root j = (-2 : R) • P.root i :=
   P.pairing_neg_one_neg_four_iff' i j <| smul_right_injective R two_ne_zero
 
+section IsValuedIn
+
+open FaithfulSMul
+
+variable [LinearOrderedCommRing S] [Algebra S R] [FaithfulSMul S R] [P.IsValuedIn S]
+omit [NoZeroSMulDivisors R N]
+variable {i j}
+
+lemma linearIndependent_iff_coxeterWeightIn_ne_four :
+    LinearIndependent R ![P.root i, P.root j] ↔ P.coxeterWeightIn S i j ≠ 4 := by
+  rw [linearIndependent_iff_coxeterWeight_ne_four, ← P.algebraMap_coxeterWeightIn S,
+    ← map_ofNat (algebraMap S R), (algebraMap_injective S R).ne_iff]
+
+lemma coxeterWeightIn_eq_four_iff_not_linearIndependent :
+    P.coxeterWeightIn S i j = 4 ↔ ¬ LinearIndependent R ![P.root i, P.root j] := by
+  rw [← coxeterWeight_eq_four_iff_not_linearIndependent, ← P.algebraMap_coxeterWeightIn S,
+    ← map_ofNat (algebraMap S R), (algebraMap_injective S R).eq_iff]
```

Comment:
Won't this also be something like
```suggestion
  rw [linearIndependent_iff_coxeterWeightIn_ne_four, not_not]
```
(untested)

Reply:
Good point, done!